src/Voron/StorageEnvironment.cs

```diff
@@ -525,15 +525,23 @@ internal LowLevelTransaction NewLowLevelTransaction(TransactionPersistentContext
                     var wait = timeout ?? (Debugger.IsAttached ? TimeSpan.FromMinutes(30) : TimeSpan.FromSeconds(30));
 
                     if (FlushInProgressLock.IsWriteLockHeld == false)
+                    {
                         flushInProgressReadLockTaken = FlushInProgressLock.TryEnterReadLock(wait);
+                        if (flushInProgressReadLockTaken == false)
+                        {
+                            GlobalFlushingBehavior.GlobalFlusher.Value.MaybeFlushEnvironment(this);
```

Comment:
I've just split the previous condition into two separate once in order to be able to throw a better error here. However I _think_ we shouldn't call `MaybeFlushEnvironment` since the flushing is already in progress.

@ayende ?

Reply:
The reason we are calling it is because we might have start flushing, had some tx in the meantime and now want to flush again, but didn't have a chance to, so this will cause us to effectively loop until we get it.

Reply:
OK